examples/apps/color_finder.py

```diff
@@ -0,0 +1,740 @@
+#!/usr/bin/env python3
+
+# Copyright (c) 2017 Anki, Inc.
+#
+# Licensed under the Apache License, Version 2.0 (the 'License');
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License in the file LICENSE.txt or at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an 'AS IS' BASIS, 
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+'''Cozmo looks around and drives after colors.
+
+When the program starts, Cozmo will look around for the color blue.
+Tap the cube illuminated blue to switch Cozmo's target color between blue, red, green, and yellow.
+Tap the blinking white cube to have the viewer display Cozmo's pixelated camera view.
+
+'''
+
+import asyncio
+import functools
+import math
+import numpy
+import sys
+
+import cozmo
+
+from cozmo.util import degrees, distance_mm, radians, speed_mmps
+from cozmo.lights import Color, Light
+try:
+    from PIL import Image, ImageColor, ImageDraw
+except ImportError:
+    sys.exit('Cannot import from PIL: Do `pip3 install --user Pillow` to install')
+    
+# map_color_to_light (dict): maps each color name with its cozmo.lights.Light value.
+# Red, green, and blue lights are already defined as constants in lights.py, 
+# but we need to define our own custom Light for yellow.
+
+map_color_to_light = {
+'green' : cozmo.lights.green_light, 
+'yellow' : Light(Color(name='yellow', rgb = (255, 255, 0))), 
+'blue' : cozmo.lights.blue_light, 
+'red' : cozmo.lights.red_light
+}
+
+# color_dict (dict): map of color names to regions in color space.
+# Instead of defining a color as a single (R, G, B) point, 
+# colors are defined with a minimum and maximum value for R, G, and B.
+# For example, a point with (R, G, B) = (40, 15, 225) falls entirely in the 'blue' region, 
+# because 0 < 40 < 50, 0 < 15 < 50, and 210 < 225 < 255.
+# A point with (R, G, B) = (95, 240, 160) does not fall exactly in one color region.
+# But applying color_distance_sqr between this color and all the colors in color_dict
+# will show that (95, 240, 160) is closest to the 'green' region.
+
+color_dict = {
+'red' : (210, 255, 0, 50, 0, 50), 
+'green' : (0, 50, 210, 255, 0, 50), 
+'blue' : (0, 50, 0, 50, 210, 255), 
+'white' : (200, 255, 200, 255, 200, 255), 
+'black' : (0, 30, 0, 30, 0, 30),
+'yellow' : (210, 255, 210, 255, 0, 50), 
+}
+
+# color_distance_sqr (fn (color, color_range) => int): returns the squared 
+# euclidean distance of color to color_range.
+
```

Comment:
This comment instead should be a docstring inside the method, like:
'''Returns the squared euclidean distance of color to color_range.'''